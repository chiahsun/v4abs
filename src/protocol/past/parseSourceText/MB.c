s0:
    if (reset)
        goto s1;
    goto s0;
    
s1:
    if (reset)
        goto s1;
    goto s2;
    
s2:
    if (reset)
        goto s1;
    if (y_valid)
        write(y);
    if (z_valid)
        write(z);
    goto s2;
