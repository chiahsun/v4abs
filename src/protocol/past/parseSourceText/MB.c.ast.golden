(init (source_text (state_statement (state_label (identifier s0) :) (transition_statement (edge_statement (if_statement_with_goto (if_statement_prefix if  ( (bool_expression (identifier reset)) )) (end_statement (goto_statement goto  (identifier s1) ;)))) (edge_statement (end_statement (goto_statement goto  (identifier s0) ;))))) (state_statement (state_label (identifier s1) :) (transition_statement (edge_statement (if_statement_with_goto (if_statement_prefix if  ( (bool_expression (identifier reset)) )) (end_statement (goto_statement goto  (identifier s1) ;)))) (edge_statement (end_statement (goto_statement goto  (identifier s2) ;))))) (state_statement (state_label (identifier s2) :) (transition_statement (edge_statement (if_statement_with_goto (if_statement_prefix if  ( (bool_expression (identifier reset)) )) (end_statement (goto_statement goto  (identifier s1) ;)))) (edge_statement (specific_update_statement (if_statement_without_goto (if_statement_prefix if  ( (bool_expression (identifier y_valid)) )) (read_or_write_or_check_statement (write_statement write ( (identifier y) ) ;)))) (specific_update_statement (if_statement_without_goto (if_statement_prefix if  ( (bool_expression (identifier z_valid)) )) (read_or_write_or_check_statement (write_statement write ( (identifier z) ) ;)))) (end_statement (goto_statement goto  (identifier s2) ;)))))))
